app/src/main/java/org/mozilla/focus/activity/MainActivity.kt

```diff
@@ -158,11 +159,14 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener {
     override fun dispatchKeyEvent(event: KeyEvent): Boolean {
         val fragmentManager = supportFragmentManager
         val browserFragment = fragmentManager.findFragmentByTag(BrowserFragment.FRAGMENT_TAG) as BrowserFragment?
+        val homeFragment = fragmentManager.findFragmentByTag(HomeFragment.FRAGMENT_TAG) as HomeFragment?
 
-        return if (browserFragment == null || !browserFragment.isVisible) {
-            super.dispatchKeyEvent(event)
+        return if (browserFragment != null && browserFragment.isVisible) {
+            browserFragment.dispatchKeyEvent(event)
```

Comment:
This still needs the `super.dispatchKey` in an || statement - browserFragment doesn't handle every case, and we always want it to fall to the super. See the code that was deleted, in comment below.